src/main/java/com/clevertap/stormdb/StormDB.java

```diff
@@ -609,11 +689,14 @@ private void iterate(final boolean useLatestWalFile, final boolean readInMemoryB
             rwLock.readLock().unlock();
         }
 
+        // first iterate over the deleteKeys file and create a set out of that and then use that set with index to check if a key should be deleted or not
+        TIntHashSet keysToBeDeleted = getKeysToBeDeletedFromFile(useLatestWalFile, readInMemoryBuffer);
```

Comment:
For every iteration, are we always reading the keys from files? Do we need to?

Reply:
If we try to keep all the deleted keys in memory then that set/map will take large memory. Also, I was not able to think of any solution other than this. if you have something in mind I am all ears.

Reply:
Hmmm, any chance that we can use a lazy bitset here?